src/main/scala/strawman/collection/mutable/ArrayBuffer.scala

```diff
@@ -39,7 +39,7 @@ class ArrayBuffer[A] private (initElems: Array[AnyRef], initLength: Int)
         start = 0
       }
       else {
-        val newelems = new Array[AnyRef](end * 2)
+        val newelems = new Array[AnyRef](if (end == 0) 16 else (end * 2))
```

Comment:
Another problem of this implementation is integer overflow. We should always look at the current collections library for implementation details where these corner cases have already been discovered and fixed.